tests/unit/disc/UnitTest_BlockedJacobian.cpp

```diff
@@ -0,0 +1,178 @@
+//*****************************************************************//
+//    Albany 3.0:  Copyright 2016 Sandia Corporation               //
+//    This Software is released under the BSD license detailed     //
+//    in the file "license.txt" in the top-level Albany directory  //
+//*****************************************************************//
+
+#include <Teuchos_ConfigDefs.hpp>
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_TimeMonitor.hpp>
+
+#include <string>
+#include <iostream>
+#include <vector>
+#include <set>
+
+#include "Panzer_BlockedDOFManager.hpp"
+#include "Panzer_IntrepidFieldPattern.hpp"
+#include "Panzer_PauseToAttach.hpp"
+
+#include "Albany_GenericSTKMeshStruct.hpp"
+#include "Albany_DiscretizationFactory.hpp"
+#include "Albany_STKDiscretization.hpp"
+#include "STKConnManager.hpp"
+#include "Albany_BlockedSTKDiscretization.hpp"
+
+// include some intrepid basis functions
+// 2D basis
+#include "Intrepid2_HGRAD_QUAD_C1_FEM.hpp"
+
+#include "Kokkos_DynRankView.hpp"
+
+#include "Albany_ThyraUtils.hpp"
+
+#include "Albany_IossSTKMeshStruct.hpp"
+
+using Teuchos::rcp;
+using Teuchos::RCP;
+using Teuchos::rcp_dynamic_cast;
+using Teuchos::rcpFromRef;
+
+typedef Kokkos::DynRankView<double, PHX::Device> FieldContainer;
+
+namespace Albany
+{
+
+   template <typename Intrepid2Type>
+   Teuchos::RCP<const panzer::FieldPattern> buildFieldPattern()
+   {
+      // build a geometric pattern from a single basis
+      Teuchos::RCP<Intrepid2::Basis<PHX::exec_space, double, double>> basis = rcp(new Intrepid2Type);
+      Teuchos::RCP<const panzer::FieldPattern> pattern = rcp(new panzer::Intrepid2FieldPattern(basis));
+      return pattern;
+   }
+
+   TEUCHOS_UNIT_TEST(AlbanyBlockedDOFManager_BlockedJacobian, assortedTests)
+   {
+
+      // Set the static variable that denotes this as a Tpetra run
+      static_cast<void>(Albany::build_type(Albany::BuildType::Tpetra));
+
+      Teuchos::RCP<const Teuchos_Comm> comm = Albany::getDefaultComm();
+
+      // panzer::pauseToAttach();
+
+      using Teuchos::RCP;
```

Comment:
Same here. I'm fine with `using foo::bar`, but then it gets confusing if I see `foo::bar` later on.

Reply:
Well, in this case it is definitely my code!
I will clean it.